adobe/copy_on_write.hpp

```diff
@@ -204,16 +194,21 @@ class copy_on_write {
 
     \return A const pointer to the underlying object
     */
-    const value_type* operator->() const { return &read(); }
+    auto operator-> () const noexcept -> const element_type* { return &read(); }
```

Comment:
Is the return type expression necessary here? It seems to me the deduced type from `read()` will suffice.